third_party/CMakeLists.txt

```diff
@@ -1,6 +1,5 @@
-if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
-  # Suppress warnings in third party subprojects.
-  add_compile_options(-Wno-everything)
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang|MSVC")
```

Comment:
Also, one thing to consider is disabling only *specific* warnings - if all of the warnings coming from `onnx` etc. fall in a couple of categories, it may be more beneficial to disable the specific warnings after verifying that they are not impactful on onnx-mlir. Warnings are supposed to have value, so it's possible that by ignoring all warnings from submodule, we miss a bug that is introduced in a newer version of the submodule and spend time investigating failures that could have easily be diagnosed based on the warnings. That's why when possible, it's always better to fix the warnings even in the submodules as @gongsu832 suggested.

Reply:
You'll still get a warning in that case - because you'll be overriding `w4` with `w0`. What llvm does is remove the old flag and add the new one.